projects/ssr-tests/src/ssr-testing.spec.ts

```diff
@@ -60,21 +63,18 @@ describe('SSR E2E', () => {
     expect(response.statusCode).toEqual(500);
   });
 
-  // TODO: Currently, the ssr server still responds with 200 quickly despite the proxy delay
-  // TODO: Test incomplete
-  xit('should receive 500 error response with timed-out request', async () => {
+  it('should receive 500 error response with timed-out request', async () => {
+    const TIMEOUT = 2000;
+    await Ssr.startSsrServer(4000, `SSR_TIMEOUT=${TIMEOUT} SSR_DEBUG=1`);
     proxy = await ProxyServer.startProxyServer({
       target: BACKEND_BASE_URL,
-      delay: 10000,
+      delay: 4000,
     });
-    const response: any = await ProxyServer.sendRequest('/timeout');
-
-    // Waits a time for server to timeout
-    await new Promise((res) => setTimeout(res, 15000));
-
-    // TODO: Assert ssr server log for timeout error
-    Log.assertMessages(['timeout']);
+    const response: any = await ProxyServer.sendRequest(REQUEST_PATH);
 
+    await Log.waitUntilLogContainsText(
+      `SSR rendering exceeded timeout ${TIMEOUT}, fallbacking to CSR for ${REQUEST_PATH}`
+    );
     expect(response.statusCode).toEqual(500);
-  }, 20000);
+  }, 10000);
 });
```

Comment:
[MEDIUM]
I believe this test doesn't match with the description.
e.g. we don't check anywhere that the reponse from SSR server is 500.
Moreover there is no chance to get 500, becasue sooner the _request_ will timeout.

I believe we want instead to:
- _increase_ the tolerated _request timeout_ to very-very long (e.g. `timeout: 40_000`)
- simulate enough long delay in backend API requests in the proxy server (e.g. `delay: 20_000`)
- (according to default Spartacus config and the implemented http timeout interceptor, the backend requests taking more than 20s are killed and turned into an error. the error should be eventually converted to setting 500 status on the SSR response)
- so we'd like to make a request, then wait ~20-25s and expect that the app will return a response with 500 status

It's too bad that Spartacus has such a big default time limit for the outgoing requests to backend API. Without changing the app's code (e.g. it's default config), we have wait that long.
Alternatively, we might want to shorten this time limit, by setting a custom value only in our storefrontapp's `app.module`, e.g. `provideConfig({ backend: { timeout: { server: 10_000 } } })`. Then it would suffice to wait ~10-15s in our tests. I think it's a fair tradeoff. The backend API requests shouldn't take more than 10s.

What do you think about it?

Reply:
These are the default options I can see in our app. Noticing max render time is 300 seconds and forced ssr timeout is 60 seconds, wouldn't we need to wait much longer than the aforementioned wait times in your suggestion? 

```
export const defaultSsrOptimizationOptions: SsrOptimizationOptions = {
  concurrency: 10,
  timeout: 3_000,
  forcedSsrTimeout: 60_000,
  maxRenderTime: 300_000,
  reuseCurrentRendering: true,
  debug: false,
  renderingStrategyResolver: defaultRenderingStrategyResolver(
    defaultRenderingStrategyResolverOptions
  )
};
```

Reply:
Ok I have just found the http-timeout-intercept with the 20 second delay. Thinking...

Reply:
I have modified the test to reflect the suggestions. How does it look to you?